P5/Exemplars/tei_jtei.odd

```diff
@@ -2525,19 +2534,17 @@
                   </sch:report>
                 </sch:rule>
                 <sch:rule context="tei:note" role="warning">
-                  <sch:report test="not(@rend='inside.parenthesis') and following::text()[not(ancestor::tei:note)][1][matches(., '^\)')]" sqf:fix="note.rend.add">
+                  <sch:report test="following::text()[not(ancestor::tei:note)][1][matches(., '^\)')]">
                     Footnotes may be placed before closing parentheses, though this is
-                    exceptionial. If this note's placement is intentional, please mark
-                    this with a @rend='inside.parenthesis' attribute. Otherwise, move
-                    it after the closing parenthesis.
+                    exceptionial. Please check if this note's placement is correct.
+                    Otherwise, move it after the closing parenthesis.
                   </sch:report>
-                  <sqf:fix id="note.rend.add">
-                    <sqf:description>
-                      <sqf:title>Add a @rend='inside.parentheses' attribute to the &lt;note&gt;.</sqf:title>
-                    </sqf:description>
-                    <sqf:add match="self::tei:note" node-type="attribute" target="rend">inside.parenthesis</sqf:add>
-                  </sqf:fix>                  
                 </sch:rule>                
+              <sch:rule context="tei:note">
+                  <sch:assert role="warning" test="matches(normalize-space(), '[.?!]$')">
```

Comment:
Don't you need to escape those characters, i.e. `[\.\?!]`?

Reply:
Not necessarily: they are interpreted as normal characters (only `\`, `[`, and `]`, and -depending on the position- `^` and `-` have to be escaped inside a character class). All other characters are treated as literal characters inside a character class. I even thought they shouldn't be escaped inside a character class, but https://www.w3.org/TR/xmlschema-2/#nt-charClassEsc seems to explain that the usual meta-characters can occur escaped inside a character class as well. In other words: `[?]` and `[\?]` are equivalent.  I prefer `[.?!]` since I think it is more readable and less confusing than `[\.\?\!]` (with non-necessary character escapes). Unless there is a general preference for the latter expression?

Reply:
No, if they don't need to be escaped, the shorter the better :) 
Thanks for the clarification, I didn't look that up